src/Parsers/ASTIdentifier.cpp

```diff
@@ -211,7 +235,7 @@ bool tryGetIdentifierNameInto(const IAST * ast, String & name)
 {
     if (ast)
     {
-        if (const auto * node = ast->as<ASTIdentifier>())
+        if (const auto * node = dynamic_cast<const ASTIdentifier *>(ast))
```

Comment:
Why dynamic cast?

Reply:
It's a shortcut - not to check `ASTIdentifier` and `ASTTableIdentifier` with the same code.